### ISSUE TITLE
Ability to add custom entries for Adventages/Disadventages, Special Abilities, Spells

### ISSUE THREAD

**Issue by ZeSandman** [elyukai/optolith-client#215]
**Is your feature request related to a problem? Please describe.**
> Indirectly, it is related to the fact that data from the books isn't all there.  
**Describe the solution you'd like**

> A Similar entry to "Create" in the Belongings or "Own Profession", but for Adventages, Disadventages, Special Abilities & Spells.

This could let us use Optolith to build characters using the latest books without having to wait for you to put all the info in.  It would also alleviate some pressure from players what want the new shiny stuff on their sheet :)

It's also useful for Fan made ideas.

**Describe alternatives you've considered**
>  Using a similiarly priced ability/spell/adventage that doesn't change statistics, and over-write entries on a printed sheet or use a PDF editor to "fix the text" of of it before printing it.

**Additional context**
>

**Comment by elyukai** [comment]
This is actually a duplicate of #28, but it's way better described here so I'm gonna keep that! ;) Thanks for your request! :)

**Comment by ZeSandman** [comment]
You're welcome... I was trying to find a way to do it simply without requiring calculation and stuff.  we can easily think to add a bonus/penalty when playing, but if it's a chore just to add then entry to the character sheet, then it's simply not used - or we're using something else to build our character

**Comment by elyukai** [comment]
The thing is: If I am gonna implement this feature, I want it to be extensible, so that you can add everything a given entry of this category has &ndash; for example, so that you can add an effect text, casting time, duration, and more to a spell. But the most important aspect is that the basics of the system needs to be integrated into the current system, which will be quite tricky, which is why I want to carefully consider every change to the current system. I don't want to break any current feature, but I also don't want to limit custom entries.

**Comment by ZeSandman** [comment]
I agree with that way of thinking, but a stop-gap measure could be useful.  have a basic way to add stuff while the full feature is being worked on.

**Comment by elyukai** [comment]
Yeah, true. Sadly, even for a MVP, there's a lot to do because I need to do the deep integration into the current system either way. And there are also open design questions like how to save custom entries, per character or per user?

**Comment by ZeSandman** [comment]
Agreed...
Save per character and it's easily exportable for someone else to import said character
Save per user... then you need a way to export those if said user has to re-install or their "saved character" might not be able to be imported...

Deep integration might take a while though... hence the idea of "Custom Entry=Text;cost=x" as a stopgap... Not elegant per-see, but sufficient for a while and better than nothing, eventually to be replaced with the "Elegant & Deeply Integrated Solution (TM)"

**Comment by elyukai** [comment]
Yeah, and I'm also thinking of this in terms of how it should work when there is a online sync *and share* later on. Entries should be able to be edited, but how to sync them between characters or even users?

Even "Custom Entry=Text;cost=x" would require a certain amount integration, because it needs to get into the system for calculating AP. And the single entries also need to be identified, which also causes problems without additional integration. I know, "Custom Entry=Text;cost=x" sounds pretty easy. But sadly, it's not. Which is why I did not implement it yet.

**Comment by n1ru4l** [comment]
Besides adding custom advantages/disadvantages it would also be nice if you could let a disadvantages reduce your AP instead of increasing them. E.g. some of our player characters got bitten by a mosquito that decreased our AP (https://de.wiki-aventurica.de/wiki/Borbarad-Moskito). 

It would be nice if something like this could either be added (via this feature) or by adding it as a new disadvantage (similar to `Furcht vor <fill-in-the-gap>`).

**Comment by ZeSandman** [comment]
Or simply letting the AP count go in the negative after character creation is "Done"...
We've hit the problem where new players wanted to tweak something, but I had to jump through hoops (reduce this and tha skill to free-up some AP, then remove this 10pt Disad to add this 10pt Disad instead, then put-back the skills where they were).

**Comment by n1ru4l** [comment]
> Or simply letting the AP count go in the negative after character creation is "Done"...

I would prefer to being able to see my total AP count and handle the subtraction for my use-case via a disadvantage. That way all players have the same AP count (despite some having less "available", due to the mosquito incident 😅).

**Comment by elyukai** [comment]
You can actually decrease your AP, but as this is very unusual and might happen unintentionally it's disabled during normal use. To be able to input a negative value when adding AP, you need to check "Edit heroes after hero creation" (aka "GM mode") in settings.

**Comment by n1ru4l** [comment]
@elyukai I think there is no official disadvantage for this in the current version of the game. Is there a rule pack that would suit this request or would that be something you would consider solving as part of this issue?

**Comment by ZeSandman** [comment]
> To be able to input a negative value when adding AP, you need to check "Edit heroes after hero creation" (aka "GM mode") in settings.

Oooh, I never thought to put in a negative number in "ADD AP"...

**Comment by elyukai** [comment]
> I think there is no official disadvantage (drained AP) for this in the current version of the game. Is there a rule pack that would suit this request or would that be something you would consider solving as part of this issue?
I think the AP drain is not an advantage/disadvantage and as such should be handled by adding negative AP. in the future, you can add a description to that, so that it's more clear why you lost AP. I'm not sure if real disadvantages that *cost* AP would really make sense, because that is basically against the idea of disadvantages: You get a penalty, but you get AP for that. While advantages are the opposite. I know, this might be sth not everyone would agree on, but in the end, Optolith is an opinionated tool, which is also built on that being said. If you like, you can create a separate issue, because I think, this is to be discussed separate from this issue.

> Oooh, I never thought to put in a negative number in "ADD AP"...
Not the best user experience, I know, but I'll draft a new UX over the next few month, which will also address issues like these when GM mode is enabled.

**Comment by Brayne-Chylde** [comment]
I was curious if there has been any movement on this subject as it is something I would be very thankful for as I am creating a setting that uses The Dark Eye rules set but a different world so it is going to require a lot of customization of professionsa and the Optolith Character tool is one of the best I've seen and my whole group uses it and we would still very much like to.

**Comment by elyukai** [comment]
Sadly no. 😕 I'm still quite busy addition functionality for new German books and addition custom entries would require *a lot* of work. If you need the same rules but different crunch elements you could customize the YAML files, which are used to define the crunch elements (they aren't publicly available, though) – you'd probably get issues with updates and custom logic for certain entries, though.
